File: Inheritance/cpp/Student.cpp
#include <string>
#include <iostream>
#include "Person.cpp"
class Student : public Person{
    private:
        float nilaiPenguji1;
        float nilaiPenguji2;
        float nilaiPembimbing;
        std::string npm;

    public:
        Student(std::string name, int age, std::string npm) : Person(name, age){
            this->npm = npm;
        }

        std::string getNpm() {
            return this->npm;
        }

        void setNpm(std::string npm) {
            this->npm = npm;
        }

        void setNilaiPenguji1(float nilaiPenguji1){
            this->nilaiPenguji1 = nilaiPenguji1;
        }

        void setNilaiPenguji2(float nilaiPenguji2){
            this->nilaiPenguji2 = nilaiPenguji2;
        }

        void setNilaiPembimbing(float nilaiPembimbing){
            this->nilaiPembimbing = nilaiPembimbing;
        }

        std::string getStatus(){
            char hurufMutu = this->getHurufMutu();

            if(hurufMutu == 'T'){
                return "Tidak Lulus !!";
            }

            return "Lulus !!";
        }

        float getNilaiPenguji1(){
            return this->nilaiPenguji1;
        }

        float getNilaiPenguji2(){
            return this->nilaiPenguji2;
        }

        float getNilaiPembimbing(){
            return this->nilaiPembimbing;
        }

        char getHurufMutu(){
            float nilaiAkhir = this->getNilaiAkhir();

            if(nilaiAkhir > 0 && nilaiAkhir <= 60){
                return 'T';
            }else if(nilaiAkhir > 60 && nilaiAkhir <= 70){
                return 'C';
            }else if(nilaiAkhir > 70 && nilaiAkhir <= 80){
                return 'B';
            }else if(nilaiAkhir > 80 && nilaiAkhir <= 100){
                return 'A';
            }

            return 'T';
        }

        float getNilaiAkhir(){
            return this->nilaiPenguji1*0.3f + this->nilaiPenguji2*0.3f + this->nilaiPembimbing*0.4f;
        }

        std::string str(){
            return "Nama : " + this->getName() + "\n" + "Umur : " + std::to_string(this->getAge()) + "\n" + "NPM : " + this->getNpm() + "\n" + "Nilai Penguji 1 : " + std::to_string(this->getNilaiPenguji1()) + "\n" + "Nilai Penguji 2 : " + std::to_string(this->getNilaiPenguji2()) + "\n" + "Nilai Pembimbing : " + std::to_string(this->getNilaiPembimbing()) + "\n" + "Nilai Akhir : " + std::to_string(this->getNilaiAkhir()) + "\n" + "Status : " + this->getStatus() + "\n" + "Huruf Mutu : " + this->getHurufMutu();
        }

        void print(){
            std::cout << "Nama: " << this->name << std::endl;
            std::cout << "Umur: " << this->age << std::endl;
            std::cout << "NPM: " << this->npm << std::endl;
            std::cout << "Status: " << this->getStatus() << std::endl;
            std::cout << "Nilai Penguji 1: " << this->nilaiPenguji1 << std::endl;
            std::cout << "Nilai Penguji 2: " << this->nilaiPenguji2 << std::endl;
            std::cout << "Nilai Pembimbing: " << this->nilaiPembimbing << std::endl;
            std::cout << "Huruf Mutu: " << this->getHurufMutu() << std::endl;
        }
};

File: Inheritance/java/Student.java
/*
  Nama File : soal-uts-1.cpp
  Nama      : Andaru Danurdara Wibisana, Fauzan Azmi Dwicahyo, Irfan Kamal, Muhammad Attila An Naufal
  NPM       : 140810200020, 140810200030, 140810200032, 140810200048
  Kelas     : B
  Tanggal   : 10 November 2021
  Deskripsi : Implementasi kasus nilai sidang menggunakan Java, class Student
*********************************************/

public class Student extends Person {
    private String npm;
    private float nilaiPenguji1;
    private float nilaiPenguji2;
    private float nilaiPembimbing;
    
    Student(String name, int age, String npm){
        super(name, age);
        this.npm = npm;
    }

    public String getNpm() {
        return this.npm;
    }

    public void setNpm(String npm) {
        this.npm = npm;
    }

    public float getNilaiPenguji1() {
        return nilaiPenguji1;
    }

    public void setNilaiPenguji1(float nilaiPenguji1) {
        this.nilaiPenguji1 = nilaiPenguji1;
    }

    public float getNilaiPenguji2() {
        return nilaiPenguji2;
    }

    public void setNilaiPenguji2(float nilaiPenguji2) {
        this.nilaiPenguji2 = nilaiPenguji2;
    }

    public float getNilaiPembimbing() {
        return nilaiPembimbing;
    }

    public void setNilaiPembimbing(float nilaiPembimbing) {
        this.nilaiPembimbing = nilaiPembimbing;
    }

    public float getNilaiAkhir(){
        return (nilaiPenguji1*0.3f + nilaiPenguji2*0.3f + nilaiPembimbing*0.4f);
    }

    public char getHurufMutu(){
        float nilai = this.getNilaiAkhir();
        if(nilai > 0 && nilai <= 60){
            return 'T';
        }else if(nilai > 60 && nilai <= 70){
            return 'C';
        }else if(nilai > 70 && nilai <= 80){
            return 'B';
        }else if(nilai > 80 && nilai <= 100){
            return 'A';
        }
        return 'T';
    }

    public String getStatus(){
        char hurufMutu = this.getHurufMutu();
        if(hurufMutu == 'T'){
            return "Tidak Lulus !!";
        }
        return "Lulus !!";
    }

    public String toString(){
        return "Nama : " + this.getName() + "\n" +
               "Umur : " + this.getAge() + "\n" +
               "NPM : " + this.getNpm() + "\n" +
               "Nilai Penguji 1 : " + this.getNilaiPenguji1() + "\n" +
               "Nilai Penguji 2 : " + this.getNilaiPenguji2() + "\n" +
               "Nilai Pembimbing : " + this.getNilaiPembimbing() + "\n" +
               "Nilai Akhir : " + this.getNilaiAkhir() + "\n" +
               "Huruf Mutu : " + this.getHurufMutu() + "\n" +
               "Status : " + this.getStatus();
    }

    public void print(){
        System.out.println(this.toString());
    }
}
